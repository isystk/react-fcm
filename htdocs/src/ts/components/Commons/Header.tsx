import React, { FC } from 'react'

export const CommonHeader: FC = () => {
  return (
    <>
      <nav className="main-header navbar navbar-expand navbar-white navbar-light">
        <ul className="navbar-nav">
          <li className="nav-item">
            <a className="nav-link" data-widget="pushmenu" href="#">
              <i className="fas fa-bars"></i>
            </a>
          </li>
        </ul>
      </nav>
    </>
  )
}

export default CommonHeader
