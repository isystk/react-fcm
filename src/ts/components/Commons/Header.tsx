import React, { FC, useState } from 'react'
import { NavDropdown, Form } from 'react-bootstrap'
import CSRFToken from '../../containers/Elements/CSRFToken'
import { URL } from '../../common/constants/url'
import { useHeader } from './Header.hooks'

export const CommonHeader: FC = () => {
  const { auths, push_top, push_login, push_register, push_mycart, push_contact } = useHeader()
  const [isSideMenu, setSideMenu] = useState(false)

  const { auth, name } = auths

  const renderLoginPc = (): JSX.Element => {
    return (
      <div className="" id="navbarSupportedContent">
        <ul className="navbar-nav ml-auto">
          {(() => {
            if (auth) {
              // ログイン済みの場合
              return (
                <>
                  <NavDropdown id="logout-nav" title={name + ' 様'}>
                    <NavDropdown.Item
                      href="/logout"
                      onClick={e => {
                        e.preventDefault()
                        const element: HTMLFormElement = document.getElementById('logout-form') as HTMLFormElement
                        if (element) {
                          element.submit()
                        }
                      }}
                    >
                      ログアウト
                    </NavDropdown.Item>
                    <Form id="logout-form" action="/logout" method="POST" style={{ display: 'none' }}>
                      <CSRFToken />
                    </Form>
                    <NavDropdown.Item href="/mycart" onClick={mycartSubmit}>
                      カートを見る
                    </NavDropdown.Item>
                    <Form id="mycart-form" action="/mycart" method="POST" style={{ display: 'none' }}>
                      <CSRFToken />
                    </Form>
                  </NavDropdown>

                  <a href={URL.MYCART} onClick={mycartSubmit}>
                    <img src="/assets/front/image/cart.png" className="cartImg ml-3" />
                  </a>
                </>
              )
            } else {
              // 未ログインの場合
              return (
                <>
                  <li className="nav-item">
                    <a
                      className="btn btn-danger mr-3"
                      href={URL.LOGIN}
                      onClick={e => {
                        e.preventDefault()
                        push_login()
                      }}
                    >
                      ログイン
                    </a>
                  </li>
                  <li className="nav-item">
                    <a
                      className="btn btn-link text-danger"
                      href={URL.REGISTER}
                      onClick={e => {
                        e.preventDefault()
                        push_register()
                      }}
                    >
                      新規登録
                    </a>
                  </li>
                </>
              )
            }
          })()}

          <li className="nav-item">
            <a
              className="btn btn-link text-danger"
              href={URL.CONTACT}
              onClick={e => {
                e.preventDefault()
                push_contact()
              }}
            >
              お問い合わせ
            </a>
          </li>
        </ul>
      </div>
    )
  }

  const renderLoginSp = (): JSX.Element => {
    const isSideMenuOpen = isSideMenu
    const sideMenuClass = isSideMenuOpen ? 'open' : ''
    const menuBtnClass = isSideMenuOpen ? 'menu-btn on' : 'menu-btn'
    const layerPanelClass = isSideMenuOpen ? 'on' : ''

    return (
      <>
        <div
          className={menuBtnClass}
          onClick={e => {
            e.preventDefault()
            const isOpen = isSideMenu ? false : true
            setSideMenu(isOpen)
          }}
        >
          <figure></figure>
          <figure></figure>
          <figure></figure>
        </div>
        <div id="side-menu" className={sideMenuClass}>
          <div className="side-menu-header">
            <p style={{ margin: '20px', fontSize: '16px' }}>{auth && name + ' 様'}</p>
          </div>
          <nav>
            <ul>
              {(() => {
                if (auth) {
                  // ログイン済みの場合
                  return (
                    <>
                      <li>
                        <a
                          href="/logout"
                          onClick={e => {
                            e.preventDefault()
                            const element: HTMLFormElement = document.getElementById('logout-form') as HTMLFormElement
                            if (element) {
                              element.submit()
                            }
                          }}
                        >
                          ログアウト
                        </a>
                        <Form id="logout-form" action="/logout" method="POST" style={{ display: 'none' }}>
                          <CSRFToken />
                        </Form>
                      </li>
                      <li>
                        <a href="/mycart" onClick={mycartSubmit}>
                          カートを見る
                        </a>
                        <Form id="mycart-form" action="/mycart" method="POST" style={{ display: 'none' }}>
                          <CSRFToken />
                        </Form>
                      </li>
                    </>
                  )
                } else {
                  // 未ログインの場合
                  return (
                    <>
                      <li>
                        <a
                          href={URL.LOGIN}
                          onClick={e => {
                            e.preventDefault()
                            setSideMenu(false)
                            push_login()
                          }}
                        >
                          ログイン
                        </a>
                      </li>
                      <li>
                        <a
                          href={URL.REGISTER}
                          onClick={e => {
                            e.preventDefault()
                            setSideMenu(false)
                            push_register()
                          }}
                        >
                          新規登録
                        </a>
                      </li>
                    </>
                  )
                }
              })()}
              <li>
                <a
                  href={URL.CONTACT}
                  onClick={e => {
                    e.preventDefault()
                    setSideMenu(false)
                    push_contact()
                  }}
                >
                  お問い合わせ
                </a>
              </li>
            </ul>
          </nav>
        </div>
        <div id="layer-panel" className={layerPanelClass}></div>
      </>
    )
  }

  const mycartSubmit = e => {
    e.preventDefault()
    setSideMenu(false)
    push_mycart()
  }

  return (
    <>
      <header className="header shadow-sm">
        <nav className="navbar navbar-expand-md navbar-light bg-white headerNav">
          <a
            className="header_logo"
            href="/"
            onClick={e => {
              e.preventDefault()
              push_top()
            }}
          >
            <img src="/assets/front/image/logo.png" alt="" className="" />
          </a>

          {renderLoginPc()}

          {renderLoginSp()}
        </nav>
      </header>
    </>
  )
}

export default CommonHeader
