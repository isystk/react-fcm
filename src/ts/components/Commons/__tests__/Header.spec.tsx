import React from 'react';
import { CommonHeader } from '../Header';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as hooks from '../Header.hooks'
import sinon from 'sinon'

describe('Header', () => {

  const stub = sinon.stub(hooks, 'useHeader');

  it('Header-Logout', () => {

    stub.returns({
      auths: {
        auth: false
      },
    });

    const wrapper = shallow(<CommonHeader/>);
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
  });


  it('Header-Login', () => {

    stub.returns({
      auths: {
        auth: true,
        name: 'テスト'
      },
    })

    const wrapper = shallow(<CommonHeader/>);
    const tree = toJson(wrapper);

    expect(tree).toMatchSnapshot();
  });

})
