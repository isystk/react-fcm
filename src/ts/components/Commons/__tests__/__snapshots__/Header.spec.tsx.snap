// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Header-Login 1`] = `
<Fragment>
  <header
    className="header shadow-sm"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-white headerNav"
    >
      <a
        className="header_logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          className=""
          src="/assets/front/image/logo.png"
        />
      </a>
      <div
        className=""
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <NavDropdown
            id="logout-nav"
            title="テスト 様"
          >
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              href="/logout"
              onClick={[Function]}
            >
              ログアウト
            </DropdownItem>
            <Form
              action="/logout"
              id="logout-form"
              inline={false}
              method="POST"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <Connect(CSRFToken) />
            </Form>
            <DropdownItem
              as={
                Object {
                  "$$typeof": Symbol(react.forward_ref),
                  "displayName": "SafeAnchor",
                  "render": [Function],
                }
              }
              disabled={false}
              href="/mycart"
              onClick={[Function]}
            >
              カートを見る
            </DropdownItem>
            <Form
              action="/mycart"
              id="mycart-form"
              inline={false}
              method="POST"
              style={
                Object {
                  "display": "none",
                }
              }
            >
              <Connect(CSRFToken) />
            </Form>
          </NavDropdown>
          <a
            href="/mycart"
            onClick={[Function]}
          >
            <img
              className="cartImg ml-3"
              src="/assets/front/image/cart.png"
            />
          </a>
          <li
            className="nav-item"
          >
            <a
              className="btn btn-link text-danger"
              href="/contact"
              onClick={[Function]}
            >
              お問い合わせ
            </a>
          </li>
        </ul>
      </div>
      <div
        className="menu-btn"
        onClick={[Function]}
      >
        <figure />
        <figure />
        <figure />
      </div>
      <div
        className=""
        id="side-menu"
      >
        <div
          className="side-menu-header"
        >
          <p
            style={
              Object {
                "fontSize": "16px",
                "margin": "20px",
              }
            }
          >
            テスト 様
          </p>
        </div>
        <nav>
          <ul>
            <li>
              <a
                href="/logout"
                onClick={[Function]}
              >
                ログアウト
              </a>
              <Form
                action="/logout"
                id="logout-form"
                inline={false}
                method="POST"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Connect(CSRFToken) />
              </Form>
            </li>
            <li>
              <a
                href="/mycart"
                onClick={[Function]}
              >
                カートを見る
              </a>
              <Form
                action="/mycart"
                id="mycart-form"
                inline={false}
                method="POST"
                style={
                  Object {
                    "display": "none",
                  }
                }
              >
                <Connect(CSRFToken) />
              </Form>
            </li>
            <li>
              <a
                href="/contact"
                onClick={[Function]}
              >
                お問い合わせ
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className=""
        id="layer-panel"
      />
    </nav>
  </header>
</Fragment>
`;

exports[`Header Header-Logout 1`] = `
<Fragment>
  <header
    className="header shadow-sm"
  >
    <nav
      className="navbar navbar-expand-md navbar-light bg-white headerNav"
    >
      <a
        className="header_logo"
        href="/"
        onClick={[Function]}
      >
        <img
          alt=""
          className=""
          src="/assets/front/image/logo.png"
        />
      </a>
      <div
        className=""
        id="navbarSupportedContent"
      >
        <ul
          className="navbar-nav ml-auto"
        >
          <li
            className="nav-item"
          >
            <a
              className="btn btn-danger mr-3"
              href="/login"
              onClick={[Function]}
            >
              ログイン
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="btn btn-link text-danger"
              href="/register"
              onClick={[Function]}
            >
              新規登録
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              className="btn btn-link text-danger"
              href="/contact"
              onClick={[Function]}
            >
              お問い合わせ
            </a>
          </li>
        </ul>
      </div>
      <div
        className="menu-btn"
        onClick={[Function]}
      >
        <figure />
        <figure />
        <figure />
      </div>
      <div
        className=""
        id="side-menu"
      >
        <div
          className="side-menu-header"
        >
          <p
            style={
              Object {
                "fontSize": "16px",
                "margin": "20px",
              }
            }
          />
        </div>
        <nav>
          <ul>
            <li>
              <a
                href="/login"
                onClick={[Function]}
              >
                ログイン
              </a>
            </li>
            <li>
              <a
                href="/register"
                onClick={[Function]}
              >
                新規登録
              </a>
            </li>
            <li>
              <a
                href="/contact"
                onClick={[Function]}
              >
                お問い合わせ
              </a>
            </li>
          </ul>
        </nav>
      </div>
      <div
        className=""
        id="layer-panel"
      />
    </nav>
  </header>
</Fragment>
`;
